JetNews/app/src/main/java/com/example/jetnews/ui/Navigation.kt

```diff
@@ -0,0 +1,124 @@
+package com.example.jetnews.ui
+
+import android.os.Bundle
+import androidx.annotation.MainThread
+import androidx.compose.getValue
+import androidx.compose.setValue
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import com.example.jetnews.ui.Screen.Article
+import com.example.jetnews.ui.Screen.Home
+import com.example.jetnews.ui.Screen.Interests
+import com.example.jetnews.ui.ScreenNames.ARTICLE
+import com.example.jetnews.ui.ScreenNames.HOME
+import com.example.jetnews.ui.ScreenNames.INTERESTS
+import com.example.jetnews.utils.getMutableStateOf
+
+/**
+ * Screen names (used for serialization)
+ */
+enum class ScreenNames { HOME, INTERESTS, ARTICLE }
+
+/**
+ * Class defining the screens we have in the app: home, article details and interests
+ */
+sealed class Screen(val id: ScreenNames) {
+    object Home : Screen(HOME)
+    object Interests : Screen(INTERESTS)
+    data class Article(val postId: String) : Screen(ARTICLE)
+}
+
+/**
+ * Helpers for saving and loading a [Screen] object to a [Bundle].
+ *
+ * This allows us to persist navigation across process death, for example caused by a long video
+ * call.
+ */
+private val SIS_SCREEN = "sis_screen"
+private val SIS_NAME = "screen_name"
+private val SIS_POST = "post"
+
+/**
+ * Convert a screen to a bundle that can be stored in [SavedStateHandle]
+ */
+private fun Screen.toBundle(): Bundle {
+    val retval = Bundle()
```

Comment:
Nit: `retval`? What does it mean? I'd prefer a better name here

Reply:
Use `bundleOf` from ktx? 

https://developer.android.com/reference/kotlin/androidx/core/os/package-summary#bundleOf(kotlin.Pair)

Reply:
Swapped for bundleOf! Thanks